components/ActionSheet/src/MDCActionSheetController.m

```diff
@@ -0,0 +1,331 @@
+
+@property(nonatomic, nullable, copy) MDCActionSheetHandler completionHandler;
+
+@end
+
+@implementation MDCActionSheetAction
+
++ (instancetype)actionWithTitle:(nonnull NSString *)title
+                          image:(nullable UIImage *)image
+                          handler:(void (^__nullable)(MDCActionSheetAction *action))handler {
+    return [[MDCActionSheetAction alloc] initWithTitle:title
+                                                 image:image
+                                               handler:handler];
+}
+
+- (instancetype)initWithTitle:(nonnull NSString *)title
+                        image:(nullable UIImage *)image
+                      handler:(void (^__nullable)(MDCActionSheetAction *action))handler {
+  self = [super init];
+  if (self) {
+    _title = [title copy];
+    _image = [image copy];
+    _completionHandler = [handler copy];
+  }
+  return self;
+}
+
+-(id)copyWithZone:(__unused NSZone *)zone {
+  MDCActionSheetAction *action = [[self class] actionWithTitle:self.title
+                                                         image:self.image
+                                                       handler:self.completionHandler];
+  return action;
+}
+
+@end
+
+@interface MDCActionSheetController () <MDCBottomSheetPresentationControllerDelegate, UITableViewDelegate>
+
+@property(nonatomic, nonnull) MDCActionSheetHeaderView *header;
+
+@property(nonatomic, nullable) MDCActionSheetListViewController *tableView;
+
+- (nonnull instancetype)initWithTitle:(nullable NSString *)title
+                              message:(nullable NSString *)message;
+
+@end
+
+@implementation MDCActionSheetController {
+  NSString *_actionSheetTitle;
+  NSMutableArray<MDCActionSheetAction *> *_actions;
+  MDCBottomSheetController *_bottomSheet;
+  MDCBottomSheetTransitionController *_transitionController;
+  BOOL mdc_adjustFontForContentSizeCategory;
+  BOOL initialLayout;
+}
+
++ (instancetype)actionSheetControllerWithTitle:(NSString *)title message:(NSString *)message {
+  MDCActionSheetController *actionSheet = [[MDCActionSheetController alloc] initWithTitle:title
+                                                                                  message:message];
+  
+  return actionSheet;
+}
+
++ (instancetype)actionSheetControllerWithTitle:(NSString *)title {
+  return [MDCActionSheetController actionSheetControllerWithTitle:title message:nil];
+}
+
+- (instancetype)init {
+  return [MDCActionSheetController actionSheetControllerWithTitle:nil message:nil];
+}
+
+- (nonnull instancetype)initWithTitle:(NSString *)title message:(NSString *)message {
+  self = [super initWithNibName:nil bundle:nil];
+  if (self) {
+    _actionSheetTitle = [title copy];
+    _message = [message copy];
+    [self commonMDCActionSheetControllerInit];
+  }
+
+  return self;
+}
+
+- (void)commonMDCActionSheetControllerInit {
+  initialLayout = false;
+  _bottomSheet = [[MDCBottomSheetController alloc] initWithContentViewController:self];
```

Comment:
I don't think you're using this property anywhere? Consider deleting.

Reply:
Done